contracts/pcv/compound/ERC4626CErc20Wrapper.sol

```diff
@@ -0,0 +1,240 @@
+        // compute exact amount of assets to send
+        uint256 exchangeRate = cToken.exchangeRateCurrent();
+        assets = shares * exchangeRate / 1e18;
+
+        // mint cTokens
+        cTokenUnderlying.safeTransferFrom(msg.sender, address(this), assets);
+        cTokenUnderlying.approve(address(cToken), assets);
+        uint256 balanceBefore = cToken.balanceOf(address(this));
+        require(cToken.mint(assets) == 0, "ERC4626CErc20Wrapper: error on cToken.mint");
+        uint256 actualShares = cToken.balanceOf(address(this)) - balanceBefore;
+        assert(shares == actualShares); // double check no rounding errors
+
+        // mint shares to receiver
+        _mint(receiver, shares);
+
+        // emit event
+        emit Deposit(msg.sender, receiver, assets, shares);
+
+        return assets;
+    }
+
+    /// @notice Redeems `shares` from `owner` and sends `assets`
+    /// of underlying tokens to `receiver`.
+    function withdraw(uint256 assets, address receiver, address owner) external returns(uint256 shares) {
+        // redeem cTokens and get the actual number of shares burnt
+        uint256 balanceBefore = cToken.balanceOf(address(this));
+        require(cToken.redeemUnderlying(assets) == 0, "ERC4626CErc20Wrapper: error on cToken.redeemUnderlying");
+        shares = balanceBefore - cToken.balanceOf(address(this));
+
+        // Check that owner approved spending on behalf of the caller
+        if (msg.sender != owner) {
+            uint256 allowed = allowance(owner, msg.sender);
+            require(allowed >= shares, "ERC4626CErc20Wrapper: spender not authorized");
+            if (allowed != type(uint256).max) {
+                _approve(owner, msg.sender, allowed - shares);
+            }
+        }
+
+        // Burn the owner's shares
+        // This checks that the owner has enough shares
+        _burn(owner, shares);
+
+        // transfer underlying to receiver
+        cTokenUnderlying.transfer(receiver, assets);
+
+        // emit event
+        emit Withdraw(msg.sender, receiver, assets, shares);
+
+        return shares;
+    }
+
+    /// @notice Redeems `shares` from `owner` and sends `assets`
+    /// of underlying tokens to `receiver`.
+    function redeem(uint256 shares, address receiver, address owner) external returns(uint256 assets) {
+        // Check that owner approved spending on behalf of the caller
+        if (msg.sender != owner) {
+            uint256 allowed = allowance(owner, msg.sender);
+            require(allowed >= shares, "ERC4626CErc20Wrapper: spender not authorized");
+            if (allowed != type(uint256).max) {
+                _approve(owner, msg.sender, allowed - shares);
+            }
+        }
+
+        // Burn the owner's shares
+        // This checks that the owner has enough shares
+        _burn(owner, shares);
+
+        // redeem cTokens
+        uint256 balanceBefore = cTokenUnderlying.balanceOf(address(this));
```

Comment:
Why would the vault hold underlying

Reply:
balance should always be zero, I'll remove this line